app/src/main/java/com/nytimes/android/sample/PersistingStoreActivity.kt

```diff
@@ -0,0 +1,77 @@
+package com.nytimes.android.sample
+
+import android.os.Bundle
+import android.support.v7.app.AppCompatActivity
+import android.support.v7.widget.LinearLayoutManager
+import android.support.v7.widget.Toolbar
+import android.util.Log
+import android.view.View
+import android.widget.Toast
+import android.widget.Toast.makeText
+import com.nytimes.android.external.store3.base.impl.BarCode
+import com.nytimes.android.external.store3.base.impl.Store
+import com.nytimes.android.sample.data.model.Post
+import com.nytimes.android.sample.data.model.RedditData
+import com.nytimes.android.sample.reddit.PostAdapter
+import com.squareup.moshi.Moshi
+import io.reactivex.Observable
+import io.reactivex.android.schedulers.AndroidSchedulers
+import io.reactivex.schedulers.Schedulers
+import kotlinx.android.synthetic.main.activity_store.*
+
+
+class PersistingStoreActivity : AppCompatActivity() {
+
+    private var postAdapter: PostAdapter? = null
+    private var persistedStore: Store<RedditData, BarCode>? = null
+    private var moshi: Moshi? = null
```

Comment:
Could we make these `lateinit` and non-nullable?

Reply:
done!